BridgeSDK/BridgeAPI/BridgeAPITypes/MachineGeneratedDoNotEdit/_SBBStudyParticipant.m

```diff
@@ -107,8 +107,13 @@ - (instancetype)initWithManagedObject:(NSManagedObject *)managedObject objectMan
     NSString *password = cacheManager.encryptionKey;
     if (password) {
         NSData *plaintext = [RNDecryptor decryptData:managedObject.ciphertext withPassword:password error:nil];
-        NSDictionary *jsonDict = [NSJSONSerialization JSONObjectWithData:plaintext options:0 error:NULL];
-        self = [self initWithDictionaryRepresentation:jsonDict objectManager:objectManager];
+        if (error && !plaintext) {
```

Comment:
? Um.... super confused by this. Where is `error` defined. It's not being used by the decryptor.

Also, shouldn't this fail if there is an error OR plaintext is nil? Otherwise, it will crash below where you are trying to serialize a `nil` value.

AND should the NSLog be wrapped in `#if DEBUG` so that you aren't submitting a password to the console log?

Reply:
ok check again?